packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -408,46 +390,37 @@ async function doGenerateArmTemplate(ctx: SolutionContext): Promise<Result<any,
     await backupExistingFilesIfNecessary(ctx);
     // Output main.bicep file
     const bicepOrchestrationFileContent = bicepOrchestrationTemplate.getOrchestrationFileContent();
-    const templateFolderPath = isMultiEnvEnabled()
-      ? path.join(ctx.root, templateFolderNew)
-      : path.join(ctx.root, baseFolder, templateFolder);
+    const templateFolderPath = path.join(ctx.root, templateFolderNew);
     await fs.ensureDir(templateFolderPath);
     await fs.writeFile(
       path.join(templateFolderPath, bicepOrchestrationFileName),
       bicepOrchestrationFileContent
     );
 
     // Output bicep module files from each resource plugin
-    const modulesFolderPath = isMultiEnvEnabled()
-      ? path.join(templateFolderPath, modulesFolder)
-      : templateFolderPath;
+    const modulesFolderPath = path.join(templateFolderPath, modulesFolder);
     await fs.ensureDir(modulesFolderPath);
     for (const module of moduleFiles) {
       // module[0] contains relative path to template folder, e.g. "./modules/frontendHosting.bicep"
       await fs.writeFile(path.join(templateFolderPath, module[0]), module[1]);
     }
 
     // Output parameter file
-    const parameterTemplateFolderPath = isMultiEnvEnabled()
-      ? path.join(ctx.root, templateFolderNew)
-      : path.join(ctx.root, baseFolder, parameterFolder);
-    const parameterTemplateFilePath = path.join(
-      parameterTemplateFolderPath,
-      parameterTemplateFileName
-    );
+    const parameterFileName = parameterFileNameTemplate.replace("@envName", ctx.envInfo.envName);
+    const parameterTemplateFolderPath = path.join(ctx.root, configsFolder);
```

Comment:
the variable name is not so accurate here, not sure whether teamsFxConfigFolder or something else is more accurate

Reply:
ok, renamed as "parameterEnvFolderPath"